src/transaction.js

```diff
@@ -121,8 +121,15 @@ export default class Transaction extends EventEmitter {
         debug('%s error running transaction query', this.txid)
       })
       .tap(() => {
-        if (status === 1) this._resolver(value)
-        if (status === 2) this._rejecter(value)
+        if (status === 1) {
+          this._resolver(value)
+        }
+        if (status === 2) {
+          if(isUndefined(value)) {
```

Comment:
Is the intention to *only* do this when rejecting with `undefined`? I'm not sure under what circumstances this issue could occur (as I haven't followed the issue closely), but other non-Error rejection values such as numbers or strings would have the same "missing stacktrace" issue.